### ISSUE TITLE
`single-use-lifetimes` false positive

### ISSUE THREAD

**Issue by BennoLossin** [rust-lang/rust#153836]
### Code

```Rust
#![deny(single_use_lifetimes)]

pub trait Foo<'a> {
    fn foo(self, h: &Handler<'a>);
}

pub struct Handler<'a>(fn(&'a i32));

pub struct Bar<'b>(&'b i32);

impl<'a, 'b: 'a> Foo<'a> for Bar<'b> {
    fn foo(self, h: &Handler<'a>) {
        (h.0)(self.0)
    }
}
```

### Current output

```text
error: lifetime parameter `'b` only used once
  --> src/main.rs:11:10
   |
11 | impl<'a, 'b: 'a> Foo<'a> for Bar<'b> {
   |          ^^ this lifetime...     -- ...is used only here
   |
```

### Desired output

(no error)

### Rationale and extra context

Removing the `'b` lifetime & replacing it with `'_` results in the following expected compile error:
```text
error: lifetime may not live long enough
  --> src/main.rs:13:9
   |
11 | impl<'a> Foo<'a> for Bar<'_> {
   |      -- lifetime `'a` defined here
12 |     fn foo(self, h: &Handler<'a>) {
   |            ---- has type `Bar<'1>`
13 |         (h.0)(self.0)
   |         ^^^^^^^^^^^^^ argument requires that `'1` must outlive `'a`
```

### Other cases

```Rust
// this impl doesn't trigger the lint, since we use `'b` twice
impl<'a, 'b> Foo<'a> for Bar<'b>
where
    'b: 'a,
{
    fn foo(self, h: &Handler<'a>) {
        (h.0)(self.0)
    }
}
```

For this reason, I think the original version without a `where` clause shouldn't trigger the lint.

### Rust Version

```Shell
rustc 1.96.0-nightly (2d76d9bc7 2026-03-09)
binary: rustc
commit-hash: 2d76d9bc76f27b03b4899e72ce561c7ac2c5cf6b
commit-date: 2026-03-09
host: x86_64-unknown-linux-gnu
release: 1.96.0-nightly
LLVM version: 22.1.0
```
